rfc/0009-rancher-integration-charts.md

```diff
@@ -0,0 +1,187 @@
+- [Example of superchart](https://github.com/rancher/charts/tree/dev-v2.6/packages/rancher-monitoring/rancher-monitoring/generated-changes/dependencies)
+
+Modify our charts to meet those. This includes (but is not exhaustive to, as it
+is a moving target):
+
+- Set minimum required Rancher chart annotations:
+  * `catalog.cattle.io/namespace`: should be postfixed with `-system` for UI
+    filtering. When creating v2 of the chart, the namespace should be different
+    than on v1, to handle upgrades/cleanup independently.
+  * `catalog.cattle.io/release-name`: If unset, release name can be changed in UI.
+  * `catalog.cattle.io/certified`
+  * `catalog.cattle.io/certified`
+  * `catalog.cattle.io/experimental`
+  * `catalog.cattle.io/ui-component`
+  * `catalog.cattle.io/display-name`
+  * `catalog.cattle.io/os`
+- Set `catalog.cattle.io/rancher-version` annotation as is required for
+  user-facing charts.
+- Provide `chart/appreadme.md`.
+- Provide logo for the chart by settings `icon` in `Chart.yaml`.
+- Rancher automatically creates a list of container images from charts for
+  airgap. Adopt Rancher's `repository` and `tag` values in values.yml.
+- Submit list of upstream images to be mirrored to
+  https://github.com/rancher/image-mirror. For normal container images, this is
+  no problem. For Wasm modules, they will not be supported by the Rancher
+  DockerHub repository.
+- Provide our own `serviceaccount`.
+- Define k8s min/max version with `kubeVersion` in `Chart.yaml`.
+- Add scheduling options at a pod (affinity) and node level
+  (taints/tolerations/selectors).
+- Provide `annotations.catalog.cattle.io/upstream-version`. This could be
+  automatically bumped on release.
+- Provide `questions.yaml` for defaults on Rancher Explorer UI.
+
+The Kubewarden stack has the following dependencies:
+- Designated main chart: `kubewarden-controller`. This will be the main charts
+  to build a Rancher package from.
+- Hard dependencies: 
+  * CRDs: in `kubewarden-crds` chart. On the main Kubewarden chart
+    (`kubewarden-controller`), set `catalog.cattle.io/auto-install:
+    kubewarden-crds=match` that points to the right CRD version.
+    This annotation accepts `match`, or a hardcoded version (e.g: `1.1.0`)
+    We ought to set it to a hardcoded version, to be able to release patch
+    versions.  of `kubewarden-controller` independently.
+  * `cert-manager`: considered a precondition of Rancher Explorer. Rancher
+    instructs users to install it prior to installing Rancher, so we can assume
+    it is always present.
+- Soft (optional) dependencies:
+  * `grafana`, `kube-prometheus-stack`: provided by `rancher-monitoring` chart.
+  * `jaeger-operator`: provided by `rancher-tracing` chart.
+  * `open-telemetry/opentelemetry-operator`: not yet in Rancher's repository. We
+    will need to submit it (possibly to
+    https://github.com/rancher/partner-charts) and maintain.
+  * `kubewarden-defaults`
```

Comment:
should we use `catalog.cattle.io/auto-install` with the `kubewarden-crds` version for `kubewarden-defaults` as we want to do for the controller?  `kubewarden-defaults` cannot be installed without `kubewarden-crds`.
We would need to somehow make sure we use the same crds version in the controller and `kubewarden-defaults`

Reply:
It is definitely an option that gets a non-broken kubewarden-defaults, but it doesnt' work for new users that don't know about kubewarden-controller.

If I understand correctly, in the end kubewarden-defaults will be marked as hidden, and it will be installed by the Rancher UI, with the logic for it inside the Rancher UI itself.